src/memory.cpp

```diff
@@ -0,0 +1,59 @@
+#include <algorithm>
+#include <iostream>
+#include "memory.hpp"
+
+
+std::array<uint8_t, 0x200000> const & Memory::getCartridge()
+{
+    return _cartridge;
+}
+
+std::array<uint8_t, 0xffff> const & Memory::getReadOnlyMemory()
+{
+    return _readOnlyMemory;
+}
+
+bool isEmpty(std::array<uint8_t, 0x200000> const & cartridge)
+{
+    for (size_t index = 0; index < cartridge.size(); index++) {
+        if (cartridge[index] != 0x0) {
+            return false;
+        }
+    }
+    return true;
+}
+bool isEmpty(std::array<uint8_t, 0xffff> const & rom)
```

Comment:
those two function are too similar, you can just use a template function. You can also simplify the code.
here an example : 
```
template<class ARRAY>
bool isEmpty(ARRAY const & memory)
{
   return std::all_of(std::begin(memory), std::end(memory),
                               [](auto elem) { return elem == 0; }); 
   // maybe the auto won't work, you can try ARRAY::value_type
}
```